Purchases/Purchasing/Purchases.swift

```diff
@@ -1294,12 +1297,64 @@ public extension Purchases {
         return await checkTrialOrIntroductoryDiscountEligibilityAsync(productIdentifiers)
     }
 
+    /**
+     * Computes whether or not a user is eligible for the introductory pricing period of a given product.
+     * You should use this method to determine whether or not you show the user the normal product price or
+     * the introductory price. This also applies to trials (trials are considered a type of introductory pricing).
+     * [iOS Introductory  Offers](https://docs.revenuecat.com/docs/ios-subscription-offers).
+     *
+     * - Note: If you're looking to use Promotional Offers instead,
+     * use ``Purchases/getPromotionalOffer(forProductDiscount:product:completion:)``.
+     *
+     * - Note: Subscription groups are automatically collected for determining eligibility. If RevenueCat can't
+     * definitively compute the eligibility, most likely because of missing group information, it will return
+     * ``IntroEligibilityStatus/unknown``. The best course of action on unknown status is to display the non-intro
+     * pricing, to not create a misleading situation. To avoid this, make sure you are testing with the latest
+     * version of iOS so that the subscription group can be collected by the SDK.
+     *
+     *
+     * - Parameter product: The ``StoreProduct``  for which you want to compute eligibility.
+     * - Parameter completion: A block that receives an ``IntroEligibilityStatus``.
+     */
+    @objc(checkTrialOrIntroDiscountEligibilityForProduct:completion:)
+    func checkTrialOrIntroDiscountEligibility(product: StoreProduct,
+                                              completion: @escaping (IntroEligibilityStatus) -> Void) {
+        trialOrIntroPriceEligibilityChecker.checkEligibility(product: product, completion: completion)
+    }
+
+    /**
+     * Computes whether or not a user is eligible for the introductory pricing period of a given product.
+     * You should use this method to determine whether or not you show the user the normal product price or
+     * the introductory price. This also applies to trials (trials are considered a type of introductory pricing).
+     * [iOS Introductory  Offers](https://docs.revenuecat.com/docs/ios-subscription-offers).
+     *
+     * - Note: If you're looking to use Promotional Offers instead,
+     * use ``Purchases/getPromotionalOffer(forProductDiscount:product:completion:)``.
+     *
+     * - Note: Subscription groups are automatically collected for determining eligibility. If RevenueCat can't
+     * definitively compute the eligibility, most likely because of missing group information, it will return
+     * ``IntroEligibilityStatus/unknown``. The best course of action on unknown status is to display the non-intro
+     * pricing, to not create a misleading situation. To avoid this, make sure you are testing with the latest
+     * version of iOS so that the subscription group can be collected by the SDK.
+     *
+     *
```

Comment:
I think we could have the Related symbols here as well, where this points to the non-async version as well, and both of them point to the dictionary version